### DISCUSSION TITLE
Dynamically retrying tests

### DISCUSSION THREAD

**Discussion by user7699** [webdriverio/webdriverio#11564] [Q&A]
Howdie! 

I'm looking for a way to dynamically retry individual tests using hooks, e.g

`    afterCommand: async function (commandName, args, result, error) {
        if(error){
            this.retries(2); //using mocha retry
        }
             },`

This (obviously) doesn't work but I hope you get the idea on what I'm trying to achieve
ERROR @wdio/utils:shim: TypeError: this.retries is not a function

I'm trying to mitigate an issue that sometimes appears when using Browserstack. Instead of re-running an enitre spec-file using the "specFileRetries" flag, I would like to look for a specific error and retry the specific test causing this issue, e.g  

`    afterCommand: async function (commandName, args, result, error) {
        if(String(error).includes('The specific error that I'm looking for')){
            this.retries(2); //using mocha retry
        }
             },`

Any input would be greatly appreciated!

**Comment by christian-bromann** [comment]
As you can see this is currently not supported. I am not sure how good of an idea it is to allow dynamic retries as it can make the behavior of the test runner very confusing. I will let others chime in on that. That said, it is certainly not impossible to implement it.

**Reply by user7699** [reply]
I know it might sound strange but in my case it actually makes sense given the specific error that I'm filtering on in the hook. I could add it (this.retry) in each individual spec-file as suggested by @sharutkarsh but I do not want to retry every test that fails. That strategy would inevitably add more time to each run rather than saving me time.  Given your answer Christian, I would assume making "this.retry()" available in the config file as ..not impossible.. ?

**Comment by sharutkarsh** [comment]
Hi 

It looks like you're trying to implement a dynamic test retry mechanism in your WebDriverIO tests using Mocha. 

Please try this https://webdriver.io/docs/retry/

Other references:
https://stackoverflow.com/questions/51413676/mocha-retry-whole-test
https://github.com/mochajs/mocha/issues/4006

If this is something specific to BrowserStack you can raise a ticket with them: https://www.browserstack.com/contact#technical-support

**Reply by user7699** [reply]
Setting "this.retry" in a spec-file isn't the problem. I would like to use it in a config-file which (as stated by Christian), is not supported.
